src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,702 @@
+    } else {
+      switch (event.key) {
+        case "Backspace":
+          this.setTime("second", "--");
+          break;
+        case "ArrowDown":
+          event.preventDefault();
+          this.decrementSecond();
+          break;
+        case "ArrowUp":
+          event.preventDefault();
+          this.incrementSecond();
+          break;
+      }
+    }
+  };
+
+  private setTime = (
+    key: "hour" | "minute" | "second" | "ampm",
+    value: number | string | AmPm
+  ): void => {
+    this.timeChanged = true;
+    switch (key) {
+      case "hour":
+        this.hour = typeof value === "number" ? formatNumberAsTimeString(value) : value;
+        break;
+      case "minute":
+        this.minute = typeof value === "number" ? formatNumberAsTimeString(value) : value;
+        break;
+      case "second":
+        this.second = typeof value === "number" ? formatNumberAsTimeString(value) : value;
+        break;
+      case "ampm":
+        if (this.hour === "--") {
+          this.ampm = value as AmPm;
+        } else {
+          const hourAsNumber = parseInt(this.hour);
+          switch (value) {
+            case "AM":
+              if (hourAsNumber >= 12) {
+                this.hour = formatNumberAsTimeString(hourAsNumber - 12);
+              }
+              break;
+            case "PM":
+              if (hourAsNumber < 12) {
+                this.hour = formatNumberAsTimeString(hourAsNumber + 12);
+              }
+              break;
+          }
+          this.ampm = value as AmPm;
+        }
+        break;
+    }
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    if (this.hourDisplayFormat === "12") {
+      this.ampm = this.getAmPm();
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render(): VNode {
+    const iconScale = this.scale === "s" || this.scale === "m" ? "s" : "m";
+    const includeSeconds = this.step !== 60 ? true : false;
+    return (
+      <Host>
+        <div class="time-picker">
+          <div>
+            <span
+              aria-label="increase hour"
+              class="hour-up shift top-left"
+              onClick={this.incrementHour}
+              onKeyDown={this.incrementHour}
+              role="button"
+              tabIndex={-1}
+            >
+              <calcite-icon icon="chevronup" scale={iconScale} />
+            </span>
+            <span
+              aria-label="Hour"
+              aria-valuemax="23"
+              aria-valuemin="1"
+              aria-valuenow={this.hour !== "--" ? parseInt(this.hour) : undefined}
+              aria-valuetext={this.hour !== "--" ? this.hour : undefined}
+              class="hour input"
```

Comment:
Suggestion: storing these in a resources file will help reuse across the component and in tests.

Reply:
Done https://github.com/Esri/calcite-components/commit/96042a6212a3501a5ec74653e497580bdd633ecf